packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -749,6 +1000,285 @@ describe('Table row selection', () => {
+                    const firstRowToSelect = 1;
+                    const lastRowToSelect = simpleTableData.length - 2;
+
+                    // Start with "lastRowToSelect" already selected
+                    await element.setSelectedRecordIds([
+                        simpleTableData[lastRowToSelect]!.id
+                    ]);
+                    await waitForUpdatesAsync();
+
+                    pageObject.clickRowSelectionCheckbox(firstRowToSelect);
+                    await selectionChangeListener.promise;
+
+                    const multiSelectListener = createEventListener(
+                        element,
+                        'selection-change'
+                    );
+                    pageObject.clickRowSelectionCheckbox(lastRowToSelect, true);
+                    await multiSelectListener.promise;
+
+                    expect(
+                        pageObject.getRowSelectionState(lastRowToSelect)
+                    ).toBe(TableRowSelectionState.selected);
+                });
+
+                it('selecting a range using shift + click does not deselect existing selection', async () => {
+                    await element.setSelectedRecordIds(['0']);
+                    await waitForUpdatesAsync();
+
+                    const firstRowToSelect = 3;
+                    const lastRowToSelect = simpleTableData.length - 2;
+                    const expectedSelection = [
+                        '0',
+                        ...simpleTableData.slice(3, -1).map(x => x.id)
+                    ];
+
+                    // Select the first row while pressing CTRL so that the initial selection isn't cleared
+                    await pageObject.clickRow(firstRowToSelect, {
+                        ctrlKey: true
+                    });
+                    await selectionChangeListener.promise;
+
+                    const multiSelectListener = createEventListener(
+                        element,
+                        'selection-change'
+                    );
+                    await pageObject.clickRow(lastRowToSelect, {
+                        shiftKey: true
+                    });
+                    await multiSelectListener.promise;
+
+                    const currentSelection = await element.getSelectedRecordIds();
+                    expect(currentSelection).toEqual(
+                        jasmine.arrayWithExactContents(expectedSelection)
+                    );
+                    expect(multiSelectListener.spy).toHaveBeenCalledTimes(1);
+                    const emittedIds = getEmittedRecordIdsFromSpy(
+                        multiSelectListener.spy
+                    );
+                    expect(emittedIds).toEqual(
+                        jasmine.arrayWithExactContents(expectedSelection)
+                    );
+                });
+
+                it('updating data during shift + click selection selects expected range if starting point is still in the data set', async () => {
```

Comment:
Are there other programmatic or interactive changes that could occur in the middle of a shift-click selection that we should have test coverage for? For example:
- programmatic or interactive sort or group operation changes the location of the previous starting row
- scrolling the table before the 2nd selection click (maybe not necessary to test this explicitly)
- programmatic changes to state like current selection or selection mode

Feel free to push back on the ROI of testing these or think of your own cases, these are just some examples.

Reply:
Added tests for shift+click along with
- sorting
- changing current selection programmatically
- changing selection mode

I did not add any tests that include scrolling. I think that would likely be low ROI, especially since the selection code doesn't interact with the virtualization code at all.